include/pybind11/detail/struct_smart_holder.h

```diff
@@ -58,6 +58,18 @@ High-level aspects:
 #include <typeinfo>
 #include <utility>
 
+// IMPORTANT: This code block must stay BELOW the #include <stdexcept> above.
+#if !defined(PYBIND11_EXPORT_GUARDED_DELETE)
+#    if defined(_LIBCPP_EXCEPTION)
```

Comment:
What does this have to do with exceptions? Copy-paste error?

Reply:
Tests pass.

I believe this is the right direction: conservative / in keeping with existing code.

Now, ideally we'd have a centralized decision that applies to both `PYBIND11_EXPORT_EXCEPTION` and `PYBIND11_EXPORT_GUARDED_DELETE`. I don't know what I'd call the central macro though, in large part because I only have very sketchy ideas what's going on under the hood. Something along the lines of: libc++ doesn't handle cross-dynamic-library `type_info` very well?

It's also unfortunate that we need to `#include <exception>` or `#include <stdexcept>` in order to make that decision. Which we shouldn't do before `#include <pybind11/conduit/wrap_include_python_h.h>` ... UGH.